qhana_plugin_runner/tasks.py

```diff
@@ -73,3 +126,31 @@ def save_task_error(self, failing_task_id: str, db_id: int):
     task_data.save(commit=True)
 
     result.forget()
+
+
+# TODO: not sure if we need save_step_error... probably fine to just use save_task_error also for errors in intermediate steps
```

Comment:
Your intuition should be correct. If a step error is recoverable then it must be handled by the plugin and may appear in the task log. If it is not recoverable then it must be escalated to a task error